Automattic/jetpack
# SEO Tools: add archive_title page title token

#### Changes proposed in this Pull Request:

* For the custom Archives page title structure in SEO Tools, add a more generic `archive_title` token which builds on the current `date` token.
* Usage of `[date]` will continue to work on a site, but we'll hide the "Date" insertion button in favor of "Archive Title".
* This will allow sites to display proper names on custom post type archives (e.g. "Projects" for the JP CPT at `/portfolio`), while still showing the date on date-based archives.
* Fixes #8918

#### Jetpack product discussion

#8918
https://github.com/Automattic/wp-calypso/pull/56084

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* With changes pulled down, on a Jetpack connected site, head to `/wp-admin/admin.php?page=jetpack#/traffic`
* Make sure `Customize your SEO settings` is enabled.
* Click on `Expand to customize the page title structures of your site.`
* For `Archives` use the buttons to insert each token presented (Site Title, Tagline, and Archive Title).
* Since we are no longer showing the `+ Date` button, manually type in `[date]`.
* You should now see an updated preview beneath the input. Both [date] and [archive_title] will show "Example Archive Title/Date" in preview.
* Click `Save settings` and refresh to make sure the custom Archive title persists.
* Check a custom post type archive page. For example you can enable Jetpack's custom Portfolio type, then view that archive at `example.com/portfolio/`. In the `<title>` you should see "Projects" replacing the `[archive_title]` and `[date]` tokens.
* Check a date archive (e.g. `example.com/2019/09`. Here you should see the appropriate date replacing the `[archive_title]` and `[date]` tokens.

Next, spin up the Calypso portion: https://github.com/Automattic/wp-calypso/pull/56084

* Navigate to `calypso.localhost:3000/marketing/traffic/example.com`
* Find the `Archives` section and make sure you see the correct custom Archives title you set in WP Admin.
* Try removing the existing `Date` and `Archive Title` token by clicking, and then adding them again.
* Customize your title a bit more and save. Then refresh and make sure it persists.
* Again check on the frontend portfolio and date-based archive pages to make sure the correct `<title>` comes through.
* To note, the Date button will only be replaced with the Archive Title button on Jetpack versions `>= 10.2-alpha`

Before:

![1](https://user-images.githubusercontent.com/15803018/132412261-829a4a2a-29e6-4463-9655-3b159ef98573.png)

After:

![2](https://user-images.githubusercontent.com/15803018/132412280-61d5a82a-e9b9-4ea4-a058-5f2b93c68f4d.png)

Calypso After:

![3](https://user-images.githubusercontent.com/15803018/132414592-c6e1fea4-9b3f-48b0-a835-ce3de51dc491.png)

Example `/portfolio` title:

> Custom Archives Title - Site Title: Jurassic Testing - Tagline: Just another WordPress site - Date: Projects - Archive Title: Projects

Example `/2021/09/` title:

> Custom Archives Title - Site Title: Jurassic Testing - Tagline: Just another WordPress site - Date: September 2021 - Archive Title: September 2021
